src/vcpkg/commands.extract.cpp

```diff
@@ -0,0 +1,166 @@
+#include <vcpkg/base/fwd/message_sinks.h>
+
+#include <vcpkg/base/files.h>
+#include <vcpkg/base/strings.h>
+
+#include <vcpkg/archives.h>
+#include <vcpkg/commands.extract.h>
+#include <vcpkg/commands.h>
+#include <vcpkg/vcpkgcmdarguments.h>
+#include <vcpkg/vcpkgpaths.h>
+
+namespace vcpkg::Commands
+{
+    static constexpr StringLiteral OPTION_STRIP = "strip";
+
+    static constexpr std::array<CommandSetting, 1> EXTRACT_SETTINGS = {{
+        {OPTION_STRIP, []() { return msg::format(msgStripOption, msg::option = "strip"); }},
+    }};
+
+    const CommandStructure ExtractCommandStructure = {
+        [] { return msg::format(msgExtractHelp); },
+        2,
+        3,
+        {{}, {EXTRACT_SETTINGS}, {}},
+        nullptr,
+    };
+
+    static StripSetting get_strip_setting(const ParsedArguments& options)
+    {
+        auto iter = options.settings.find(OPTION_STRIP);
+        if (iter != options.settings.end())
```

Comment:
```suggestion
        if (iter == options.settings.end())
```

Maybe consider inverting the condition to avoid this long branch?